synapse/storage/databases/main/roommember.py

```diff
@@ -759,26 +761,31 @@ async def _get_joined_hosts(
         # on it. However, its important that its never None, since two current_state's
         # with a state_group of None are likely to be different.
         assert state_group is not None
+        assert state_entry.state_group is None or state_entry.state_group == state_group
```

Comment:
I am surprised that it is ok for `state_entry.state_group` to be None when `state_group` is not None.  What's going on there? (Can you write a comment that explains it?)

Reply:
... I now grok this, having read the rest of the code, but if this is staying, it needs a PR.